config/config.go

```diff
@@ -483,13 +483,48 @@ func (c *Configuration) loadGitConfig() {
 	}
 }
 
+func filterString(r rune) rune {
+	switch r {
+	case '<', '>', '\n':
+		return -1
+	default:
+		return r
+	}
+}
+
+// findUserData returns the name/email that should be used in the commit header.
+// We use the same technique as Git for finding this information, except that we
+// don't fall back to querying the system for defaults if no values are found in
+// the Git configuration or environment.
+//
+// envType should be "author" or "committer".
+func (c *Configuration) findUserData(envType string) (name, email string) {
+	envType = strings.ToUpper(envType)
+
+	name, ok := c.Os.Get("GIT_" + envType + "_NAME")
+	if !ok {
+		name, _ = c.Git.Get("user.name")
+	}
+
+	email, ok = c.Os.Get("GIT_" + envType + "_EMAIL")
+	if !ok {
+		email, ok = c.Git.Get("user.email")
+	}
+	if !ok {
+		email, _ = c.Os.Get("EMAIL")
+	}
+
+	// Git filters certain characters out of the name and email fields.
+	name = strings.Map(filterString, name)
```

Comment:
Nice use of `strings.Map`. I think that `filterString` is too generically named to be exposed privately at the package-level scope. What do you think about either clarifying the name (perhaps, `filterSignatureString`) or introducing the function within this function scope, like:

```go
func (c *Configuration) findUserData(envType string) (name, email string) {
        var filter func(rune) rune = func(r rune) rune {
                // ...
        }

        name = strings.Map(filter, name)
}
```

Reply:
Yeah, I was worried about that.  I think the nested function is the way to go.  I'll make that change.